### PR TITLE
Automatically generate fuzz helpers

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Move fuzz helpers to `resim/converter/` and start implementing automatic generation of fuzz helpers for trivial algebraic types (i.e. types which are just a collection of fuzzable types).

## Guide to reproduce test results.
```bash
bazel test //resim/converter:parser_test
bazel test //resim/converter:fuzz_helpers_test
bazel test //resim/utils:setter_reference_test
bazel test //resim/msg:fuzz_helpers_test
```
## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: resim/converter/fuzz_helpers_test.cc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#include <vector>
+
+#include "resim/converter/fuzz_helpers_test_template.hh"
+#include "resim/converter/parser.hh"
+#include "resim/converter/testing/test_message.pb.h"
+#include "resim/time/timestamp.hh"
+
+namespace resim::converter {
+
+namespace testing {
```
Comment (iainjwhiteside) [inline resim/converter/fuzz_helpers_test.cc:20]: nice and compact!

File: resim/converter/parser.hh [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  constexpr explicit Parser(Getters getters) : getters_{std::move(getters)} {}
+
+  static constexpr std::size_t NUM_FIELDS = std::tuple_size_v<Getters>;
+
+  template <std::size_t Idx>
+  using FieldType =
+      std::remove_cv_t<std::remove_reference_t<decltype(std::get<Idx>(
+          std::declval<Getters>())(std::declval<const T &>()))>>;
+
+  // The actualy getter
```
Comment (iainjwhiteside) [inline resim/converter/parser.hh:34]:
```suggestion
  // The actual getter
```

**Review by iainjwhiteside — APPROVED** [review]
with very little confidence! in the parser stuff

File: resim/utils/setter_reference.hh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  // Operator= which calls the given setter
+  template <typename... Args>
+  SetterReference<Setter, Getter> &operator=(Args &&...args) {
+    setter_(std::forward<Args>(args)...);
+    return *this;
+  }
+
+  // Implicit conversion to the type returned by the getter.
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  operator decltype(std::declval<Getter>()())() const { return getter_(); }
```
[the remark this answers is not in the document]
Comment (mikebauer) [reply resim/utils/setter_reference.hh:33]: 😬

File: resim/converter/parser.hh (hunk 2 above)
Comment (mikebauer) [reply resim/converter/parser.hh:34]: Good catch. Fixed!

**Pull request merged**